homeassistant/components/lamarzocco/lm_client.py

```diff
@@ -0,0 +1,190 @@
+"""La Marzocco Cloud API client."""
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from lmcloud import LMCloud
+from lmcloud.exceptions import BluetoothConnectionFailed
+
+from homeassistant.components import bluetooth
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MAC,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    CONF_MACHINE,
+    MACHINE_NAME,
+    MODEL_GS3_AV,
+    MODEL_GS3_MP,
+    MODEL_LM,
+    MODEL_LMU,
+    SERIAL_NUMBER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+MODELS = [MODEL_GS3_AV, MODEL_GS3_MP, MODEL_LM, MODEL_LMU]
+
+
+class LaMarzoccoClient(LMCloud):
```

Comment:
Most of the code can be in the third-party library and should be there. Please move it there.

If I understood your code correct, you need only a Bluetooth scanner and a callable for `bluetooth.async_register_callback` from HA.

Reply:
Moved any code out that is not directly calling HA functionality, or that is only used in HA context and not needed in the 3rd party lib.

Reply:
not related to websocket, but to bluetooth mainly as most of those things rely on the `bluetooth` integration to work. I just had the websocket stuff here to use `hass.async_create_task`, if I use asyncio to spawn the task in the 3rd party lib, I can move anything websocket related there.